### ISSUE TITLE
forum topic access rights

### ISSUE THREAD

**Issue by kayjay07** [dotproject/dotProject#75]
Is there any way we can control the access right for the user not to delete the topics under the forum module. Currently, it seems the user can delete their topics.

**Comment by Givo29** [comment]
Currently there are three cases for which a user can edit or delete a forum/message:

1. The forum creator can delete their forum
2. Administrators can delete and edit all forums and messages
3. Message authors can edit and delete their own messages.   

From what I can see there is no way to change that for the moment, what kind of functionality are you looking for?

**Comment by kayjay07** [comment]
Thanks for your reply. Much appreciated. Actually, I was aiming for functionality that user should not delete their message if a reply is posted to that message avoiding the break in the transcripts loop.

**Comment by Givo29** [comment]
No problems,

I see what you mean, when a user deletes their message any replies to it are moved up.

I can look into how this could be changed: Its not too hard to simply restrict a comment author from deleting their comment but that won't solve the problem of an administrator doing the same thing. I'll have a look and see if it would be possible to check for replies on delete and possibly either delete all replies or notify any users that the original comment has been deleted.

**Comment by Givo29** [comment]
Yeah I think that would be a good solution for it, once I get a test server running at home I can take a proper look at how I might be able to implement it
